code/modules/paperwork/papershredder.dm

```diff
@@ -92,8 +125,40 @@
 	paperamount--
 	return PoolOrNew(/obj/item/weapon/shreddedp, get_turf(src))
 
+/obj/machinery/papershredder/power_change()
+	..()
+	spawn(rand(0,15))
```

Comment:
Several other machines have this random delay (fire, air alarms etc).   It makes it so when power is restored to a room full of stuff it all flickers to life instead of instant-on.  Which is cool and cinematic, so I copied it.